pkg/datasources/secrets_acceptance_test.go

```diff
@@ -0,0 +1,305 @@
+					assert.Check(resource.TestCheckTypeSetElemAttr(dsName, "secrets.2.describe_output.0.oauth_scopes.*", "first_scope")),
+					assert.Check(resource.TestCheckTypeSetElemAttr(dsName, "secrets.2.describe_output.0.oauth_scopes.*", "second_scope")),
+
+					assert.Check(resource.TestCheckResourceAttr(dsName, "secrets.3.describe_output.0.name", idFour.Name())),
+					assert.Check(resource.TestCheckResourceAttr(dsName, "secrets.3.describe_output.0.database_name", idFour.DatabaseName())),
+					assert.Check(resource.TestCheckResourceAttr(dsName, "secrets.3.describe_output.0.schema_name", idFour.SchemaName())),
+					assert.Check(resource.TestCheckResourceAttr(dsName, "secrets.3.describe_output.0.secret_type", "OAUTH2")),
+					assert.Check(resource.TestCheckResourceAttrSet(dsName, "secrets.3.describe_output.0.oauth_refresh_token_expiry_time")),
+				),
+			},
+		},
+	})
+}
+
+func secretsDataAndVars() string {
+	return `
+variable "like" {
+  type = string
+}
+
+data "snowflake_secrets" "test" {
+  depends_on = [snowflake_secret_with_basic_authentication.test, snowflake_secret_with_generic_string.test, snowflake_secret_with_client_credentials.test, snowflake_secret_with_authorization_code_grant.test]
+  like = var.like
+}
+
+`
+}
+
+func secretsData(secretResourceName string) string {
+	return fmt.Sprintf(`
+data "snowflake_secrets" "test" {
+  depends_on = [%s.test]
+}`, secretResourceName)
+}
+
+func TestAcc_Secrets_WithClientCredentials(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+
+	integrationId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	_, apiIntegrationCleanup := acc.TestClient().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "test_oauth_client_id", "test_oauth_client_secret").
+			WithOauthAllowedScopes([]sdk.AllowedScope{{Scope: "username"}, {Scope: "test_scope"}}),
+	)
+	t.Cleanup(apiIntegrationCleanup)
+
+	secretModel := model.SecretWithClientCredentials("test", integrationId.Name(), id.DatabaseName(), id.SchemaName(), id.Name(), []string{"username", "test_scope"})
+
+	dataSecretsClientCredentials := accConfig.FromModel(t, secretModel) + secretsData(secretWithClientCredentials)
+
+	dsName := "data.snowflake_secrets.test"
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.SecretWithClientCredentials),
+		Steps: []resource.TestStep{
+			{
+				Config: dataSecretsClientCredentials,
+				Check: assert.AssertThat(t,
+					assert.Check(resource.TestCheckResourceAttr(dsName, "secrets.#", "1")),
+
+					assert.Check(resource.TestCheckResourceAttr(dsName, "secrets.0.show_output.#", "1")),
```

Comment:
there's a trick to use show_output asserts here, check `users_datasource_show_output_ext.go`

Reply:
Will use that